pkg/operator/ceph/cluster/controller.go

```diff
@@ -527,8 +527,37 @@ func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID
 	return false, nil
 }
 
-// removeFinalizer removes a finalizer
-func removeFinalizer(client client.Client, name types.NamespacedName) error {
+func removeFinalizers(client client.Client, cephCluster *cephv1.CephCluster, name types.NamespacedName) error {
+	// Remove cephcluster finalizer
+	err := removeCephClusterFinalizer(client, name)
+	if err != nil {
+		return errors.Wrap(err, "failed to remove cephcluster finalizer")
+	}
+
+	// Remove monSecretsFinalizer
+	err = RemoveMonSecretFinalizer(client, cephCluster.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to remove mon secret finalizer")
+	}
+
+	// Remove monConfigMapFinalizer
+	err = RemoveMonConfigMapFinalizer(client, cephCluster.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to remove mon configmap finalizer")
+	}
+	return nil
+}
+
+func removeFinalizer(client client.Client, object client.Object) error {
```

Comment:
I'd suggest no need for this method. You can just call `opcontroller.RemoveFinalizer(client, object)` directly, right?